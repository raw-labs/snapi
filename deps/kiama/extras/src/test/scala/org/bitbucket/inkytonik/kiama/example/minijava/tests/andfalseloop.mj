class AndFalseLoop {
    public static void main () {
        System.out.println (false && (new LoopClass ().run ()));
    }
}

class LoopClass {

    public boolean run () {
        while (true) {}
        return false;
    }

}
