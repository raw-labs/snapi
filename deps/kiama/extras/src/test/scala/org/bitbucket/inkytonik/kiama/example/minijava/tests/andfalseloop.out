.source src/test/scala/org/bitbucket/inkytonik/kiama/example/minijava/tests/andfalseloop.mj
.class public AndFalseLoop
.super java/lang/Object

.method <init>()V
.limit stack 1
.limit locals 1
    aload_0
    invokespecial java/lang/Object/<init>()V
    return
.end method

.method public static main([Ljava/lang/String;)V
.limit stack 3
.limit locals 2
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    ifeq L1
    new LoopClass
    dup
    invokespecial LoopClass/<init>()V
    invokevirtual LoopClass/run()Z
    goto L2
L1:
    iconst_0
L2:
    invokevirtual java/io/PrintStream/println(I)V
    return
.end method

.source src/test/scala/org/bitbucket/inkytonik/kiama/example/minijava/tests/andfalseloop.mj
.class public LoopClass
.super java/lang/Object

.method <init>()V
.limit stack 1
.limit locals 1
    aload_0
    invokespecial java/lang/Object/<init>()V
    return
.end method

.method public run()Z
.limit stack 1
.limit locals 1
L1:
    iconst_1
    ifeq L2
    goto L1
L2:
    iconst_0
    ireturn
.end method

